### PR TITLE
 [Term Entry] Add Python Plotly- graph_objects .Surface()

### LINKED ISSUE #5179: [Term Entry] Python Plotly- graph_objects .Surface()
### Reason/inspiration (optional)

We would like a new term entry on the `.Surface()` method of the [`graph_objects module`](https://www.codecademy.com/resources/docs/plotly/graph-objects) in Plotly, Python. The entry should go in a new file under `docs/content/plotly/concepts/graph_objects/terms/surface/surface.md`

The entry should include the following:

- A description of the term
- A ##Syntax section that includes the syntax of the method and its details
- An ##Example section that includes a ```py ``` codeblock and ```shell ``` that shows the example output.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by saldanhad**
### Description

Added documentation for the `.Surface()` method of the graph_objects module in Plotly.

### Issue Solved

Closes #5179 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/plotly/concepts/graph-objects/terms/surface/surface.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    x=None,
+    y=None,
+    colorscale=None,
+    cmin=None,
+    cmax=None,
+    opacity=None,
+    surfacecolor=None,
+    **kwargs
+)
+```
```
Comment (SaviDahegaonkar) [inline content/plotly/concepts/graph-objects/terms/surface/surface.md:34]:
```suggestion
```pseudo
plotly.graph_objects.Surface(
    z=None,
    x=None,
    y=None,
    colorscale=None,
    cmin=None,
    cmax=None,
    opacity=None,
    surfacecolor=None,
    **kwargs
)
```

```
The syntax in docs is written inside the ```pseudo ``` blocks, not the ```python``` blocks. You can have a look at the [style-guide ](documentation/style-guide.md) for better information and understanding.

File: content/plotly/concepts/graph-objects/terms/surface/surface.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+- `z` : 2D array-like, required. The Z coordinates for the surface plot.
+- `x` : 1D array-like, optional. The X coordinates. Defaults to the range of the number of columns in z if not provided.
+- `y` : 1D array-like, optional. The Y coordinates. Defaults to the range of the number of rows in z if not provided.
+- `colorscale` : list of tuples, optional. Colorscale for the surface.
+- `cmin` : float, optional. Minimum value for the color scale.
+- `cmax` : float, optional. Maximum value for the color scale.
+- `opacity` : float, optional. Opacity of the surface.
+- `surfacecolor` : array-like, optional. An array of values to color the surface.
```
Comment (SaviDahegaonkar) [inline content/plotly/concepts/graph-objects/terms/surface/surface.md:43]:
```suggestion
- `z` : 2D array-like, required. The Z coordinates for the surface plot.
- `x` : 1D array-like, optional. The X coordinates. Defaults to the range of the number of columns in z if not provided.
- `y` : 1D array-like, optional. The Y coordinates. Defaults to the range of the number of rows in z if not provided.
- `colorscale` : list of tuples, optional. Colorscale for the surface.
- `cmin` : float, optional. Minimum value for the color scale.
- `cmax` : float, optional. Maximum value for the color scale.
- `opacity` : float, optional. The opacity of the surface.
- `surfacecolor` : array-like, optional. An array of values to color the surface.

```

File: content/plotly/concepts/graph-objects/terms/surface/surface.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Example
+
+The following example showcases the use of the `.Surface()` function:
```
Comment (SaviDahegaonkar) [inline content/plotly/concepts/graph-objects/terms/surface/surface.md:47]:
```suggestion
The following example showcases the use of the `Surface` class :

```
Surface is a class and not a function.

File: content/plotly/concepts/graph-objects/terms/surface/surface.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                      xaxis_title='X',
+                      yaxis_title='Y',
+                      zaxis_title='f(x, y)'
+                  ))
+
+
+fig.show()
+```
+
+This example allows for a visual representation of a mathematical function, with customizable parameters like the colorscale to highlight different levels on the surface. The z values create the height of the surface, while x and y form the base grid, enabling an intuitive understanding of how the function behaves across the input space.
```
Comment (SaviDahegaonkar) [inline content/plotly/concepts/graph-objects/terms/surface/surface.md:71]:
```suggestion
This example allows for a visual representation of a mathematical function, with customizable parameters like the color scale to highlight different levels on the surface. The z values create the height of the surface, while x and y form the base grid, enabling an intuitive understanding of how the function behaves across the input space.

```

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey @saldanhad ,
The entry is nicely written. Though I have suggested you some changes please make them asap.

**Comment by saldanhad** [comment]
Hi @SaviDahegaonkar, thank you for your review and apologies for missing a few changes earlier. Please refer to the last commit for the updated changes.

**Comment by saldanhad** [comment]
Hi @SaviDahegaonkar, apologize for the multiple commits here due to some errors from my end, however I have made the changes suggested by you, for the sake of simplicity do you prefer me closing this PR and generating a new PR for this issue ?

**Comment by SaviDahegaonkar** [comment]
Hello @saldanhad ,
As we discussed, I am closing this OR for now. You can create a new PR on this entry. Let me know if you have any more questions.

**Comment by saldanhad** [comment]
Hi @SaviDahegaonkar, I believe there might be some miscommunication, may I know what we discussed on and why was this PR closed ?

**Comment by SaviDahegaonkar** [comment]
I had got a mail from your side to close the PR and let you create a new PR
on the same issue with the specific edits.

**Comment by saldanhad** [comment]
Hi @SaviDahegaonkar, Thanks for clarifying, will raise a new PR for the same.

**Pull request closed without merging**